v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/common/ISourceDao.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.common;
+
+public interface ISourceDao {
+
+  /**
+   * Executes a given write statement.
+   *
+   * @param statement The SQL or query statement.
+   * @throws Exception If there is an error executing the statement.
+   */
+  void write(String statement) throws Exception;
```

Comment:
Will it always be this? If this is an interface should you use generics to implement this so that concrete implementations can define their own `I/O` models?